src/reducers/index.js

```diff
@@ -0,0 +1,30 @@
+const initState = { message: "Działa!", time: new Date(), users: [] };
+
+const reducer = (state = initState, action) => {
+  switch (action.type) {
+    case "getCurrentTime":
+      return {
+        ...state,
+        time: new Date(),
+      };
+    case "addUser":
+      return {
+        ...state,
+        users: [...state.users, action.payload],
+      };
+    case "deleteUser":
+      return {
+        ...state,
+        users: action.payload,
+      };
```

Comment:
Wygodniej i "ładniej" będzie tutaj wykonać ten "filter" czyli `users: state.users.filter(u => u.id != action.payload)` :)
Wtedy komponent się nam odchudzi i nie będziemy sobie zaprzątać głowy w nim jak to zrobić. Mamy akcję i ją wykonujemy. Co się dzieje pod spodem jest mniej istotne ;)